src/server/pfs/server/api_server_v2.go

```diff
@@ -148,34 +183,45 @@ func (a *apiServer) GetTarConditionalV2(server pfs.API_GetTarConditionalV2Server
 			if err := w.Flush(); err != nil {
 				return err
 			}
-			return server.Send(&pfs.GetTarConditionalResponseV2{EOF: true})
+			return server.Send(&pfs.GetTarConditionalResponse{EOF: true})
 		})
 		return bytesWritten, err
 	})
 }
 
 type getTarConditionalWriter struct {
-	server       pfs.API_GetTarConditionalV2Server
+	server       pfs.API_GetTarConditionalServer
 	bytesWritten int64
 }
 
-func newGetTarConditionalWriter(server pfs.API_GetTarConditionalV2Server) *getTarConditionalWriter {
+func newGetTarConditionalWriter(server pfs.API_GetTarConditionalServer) *getTarConditionalWriter {
 	return &getTarConditionalWriter{
 		server: server,
 	}
 }
 
 func (w *getTarConditionalWriter) Write(data []byte) (int, error) {
-	if err := w.server.Send(&pfs.GetTarConditionalResponseV2{Data: data}); err != nil {
+	if err := w.server.Send(&pfs.GetTarConditionalResponse{Data: data}); err != nil {
 		return 0, err
 	}
 	w.bytesWritten += int64(len(data))
 	return len(data), nil
 }
 
-func (a *apiServer) ListFileV2(req *pfs.ListFileRequest, server pfs.API_ListFileV2Server) error {
+func (a *apiServerV2) ListFileV2(req *pfs.ListFileRequest, server pfs.API_ListFileV2Server) error {
 	pachClient := a.env.GetPachClient(server.Context())
 	return a.driver.listFileV2(pachClient, req.File, req.Full, req.History, func(finfo *pfs.FileInfoV2) error {
 		return server.Send(finfo)
 	})
 }
+
+// FinishCommitInTransaction is identical to FinishCommit except that it can run
+// inside an existing etcd STM transaction.  This is not an RPC.
+func (a *apiServerV2) FinishCommitInTransaction(
```

Comment:
Does this work? It seems like a FinishCommit call on *apiServerV2 would get dispatched to the embedded *apiServer and the subsequent calls would have *apiServer as the receiver, not the "parent" *apiServerV2.

Reply:
I think any V2 code changes will require the rpc function to be attached to *apiServerV2, otherwise you would be making changes to *apiServer which I think is what you want to avoid with this.

Reply:
Replying here for visibility: Because this calls another rpc method it starts from the external interface (in this case APIServer).  The RealEnv setup passes an APIServer (the interface not the implementation) to txnEnv